packages/patternfly-4/react-core/src/declarations.d.ts

```diff
@@ -3,4 +3,18 @@
 declare module '@patternfly/patternfly/components/*.css' {
   const styles: any;
   export default styles;
-}
\ No newline at end of file
+}
+
+// We use a JS babel plugin to resolve these imports that `tsc` doesn't
+// know about, so pretend that everything will work
+declare module '@patternfly/patternfly/layouts/*.css' {
```

Comment:
I like this solution, but I think my PR is about to be merged (Alert) that just changes it to `declare module '@patternfly/patternfly/*/*.css'`, so you might want to go with that or deal with merge conflicts later.